src/storage/conversion/GenericRdfToQuadConverter.ts

```diff
@@ -0,0 +1,50 @@
+import { PassThrough, Readable } from 'stream';
+import rdfParser from 'rdf-parse';
+import { Representation } from '../../ldp/representation/Representation';
+import { RepresentationMetadata } from '../../ldp/representation/RepresentationMetadata';
+import { CONTENT_TYPE_QUADS, DATA_TYPE_QUAD } from '../../util/ContentTypes';
+import { UnsupportedHttpError } from '../../util/errors/UnsupportedHttpError';
+import { checkRequest } from './ConversionUtil';
+import { RepresentationConverter, RepresentationConverterArgs } from './RepresentationConverter';
+
+/**
+ * Converts most major RDF serializations to `internal/quads`.
+ */
+export class GenericRdfToQuadConverter extends RepresentationConverter {
+  public async canHandle(input: RepresentationConverterArgs): Promise<void> {
+    checkRequest(input, await rdfParser.getContentTypes(), [ CONTENT_TYPE_QUADS ]);
+  }
+
+  public async handle(input: RepresentationConverterArgs): Promise<Representation> {
+    return this.turtleToQuads(input.representation, input.identifier.path);
+  }
+
+  private turtleToQuads(representation: Representation, baseIRI: string): Representation {
+    const metadata: RepresentationMetadata = { ...representation.metadata, contentType: CONTENT_TYPE_QUADS };
+
+    // Catch parsing errors and emit correct error
+    // Node 10 requires both writableObjectMode and readableObjectMode
+    const errorStream = new PassThrough({ writableObjectMode: true, readableObjectMode: true });
+    let data: Readable;
+    try {
+      data = rdfParser.parse(representation.data, {
+        ...representation.metadata.contentType ?
+          { contentType: representation.metadata.contentType } :
+          { path: baseIRI },
+        baseIRI,
+      });
+      data.pipe(errorStream);
+      data.on('error', (error): boolean => errorStream.emit('error', new UnsupportedHttpError(error.message)));
+    } catch (error) {
+      data = errorStream;
+      setImmediate((): boolean => data.emit('error',
+        new UnsupportedHttpError('Could not find a valid content type or known file extension.')));
```

Comment:
If this catch can only occur when there is no content-type I would also drop this part then (since it means there is an internal server problem if we get to this point).